pkg/sql/schemachanger/scplan/internal/rules/registry.go

```diff
@@ -36,6 +36,9 @@ func (r *Registry) ApplyDepRules(ctx context.Context, g *scgraph.Graph) error {
 		start := timeutil.Now()
 		var added int
 		if err := dr.q.Iterate(g.Database(), func(r rel.Result) error {
+			if ctx.Err() != nil {
```

Comment:
nice! do we still need the check on line 54? and also, do we know why that check wasn't good enough?

Reply:
same, we are iterating over all dep rules, and already check for cancellation between each dep rule. The change made here simply says "also check for cancellation while applying a dep rule". I wonder how much help this will be.

Reply:
ah yeah ok i do believe we do not need this check on line 54 now. 

this check was added as a "could it be possible that we do need to check for cancellation while applying a dep rule" & after these changes were made, we (@fqazi and i) saw that the schemaChangeColumn tests did not fail (random syntax did?) - leading me to believe that we possibly got stuck applying a dep rule which led us to never be able to run the check on line 54

does that sound plausible? 

what are our stances of marking this commit as an "informs" instead of a fix to see what nightlies say

Reply:
@Xiang-Gu While I was looking at a different issue involving DROP DATABASE being super slow, I did notice it was possible to get rules which take a long time to evaluate (https://github.com/cockroachdb/cockroach/issues/100850). There are some numbers in that issue, and I don't think those are from the full scale either.

Reply:
Could you clarify this part:
> we (@fqazi and i) saw that the schemaChangeColumn tests did not fail (random syntax did?)

from what i understand, there is just one test failing: `TestRandomSyntaxSchemaChangeColumn`

Reply:
yes, sorry! for background context: we can repro this failure only when we run the whole RandomSyntax testing suite. i have had no luck reproing TestRandomSyntaxSchemaChangeColumn standalone. a repro is:
```
  ./dev test pkg/sql/tests/... --filter=TestRandomSyntax --test-args="-rsg-routines=8 -rsg-exec-timeout=1m -rsg=5m"
```

where if we check the stack traces in the build log, we normally see applyDepRules in the trace of a stuck connection (similar to what we see with CI failures). after adding the check for any cancellations while applying a dep rule, the build log errors pointed us to RandomSyntax failing *sometimes*

Reply:
thanks for explaining more! ok, i'm fine to merge this if we do the two thing discussed in this thread:
- remove the redundant check on line 54
- change this to "informs" and let's see how the tests behave over the weekend